examples/md-flexible/CMakeLists.txt

```diff
@@ -18,20 +18,20 @@ target_link_libraries(md-flexible PUBLIC autopas autopasTools molecularDynamicsL
 
 # Set CMake option for md-flexible mode
 set(MD_FLEXIBLE_MODE
-        "SINGLE-SITE"
+        "SINGLESITE"
         CACHE
         STRING "Choose the mode of md-flexible to be compiled."
         )
-set_property(CACHE MD_FLEXIBLE_MODE PROPERTY STRINGS "SINGLE-SITE;MULTI-SITE")
+set_property(CACHE MD_FLEXIBLE_MODE PROPERTY STRINGS "SINGLESITE;MULTISITE")
 
 # Define numerical values for different md-flexible modes
 target_compile_definitions(md-flexible PUBLIC SINGLESITE=0 )
 target_compile_definitions(md-flexible PUBLIC MULTISITE=1)
 
 # Define numerical value for actually compiled md-flexible mode, as per above.
-if (MD_FLEXIBLE_MODE MATCHES "SINGLE-SITE")
+if (MD_FLEXIBLE_MODE MATCHES "SINGLESITE|singlesite")
     target_compile_definitions(md-flexible PUBLIC MD_FLEXIBLE_MODE=0)
-elseif(MD_FLEXIBLE_MODE MATCHES "MULTI-SITE")
+elseif(MD_FLEXIBLE_MODE MATCHES "MULTISITE|multisite")
```

Comment:
Same as above

Reply:
But is this really necessary? Shouldn't md-flexible link against the application library and thus receive all public definitions? (Can't currently check if this is the case)

Reply:
Yes that works, I just tested it. I could move all the `target_compile_definitions` for `MD_FLEXIBLE_MODE` into the CMakeLists.txt of applicationLibrary. 
However, I would find it nicer if the md-flexible related variables would stay in the CMakeLists.txt of md-flexible. But so far I haven't found a way to set the `MD_FLEXIBLE_MODE` variable in the applicationLibrary without setting the `target_compile_definitions` in the CMakeLists.txt of applicationLibrary again. 
This variable was not set so far for the applicationLibrary and therefore all the warnings appeared even if md-flexible was compiled with multisite support.

(But if moving the definitions to applicationLibrary is the only possibility, then of course I can do that.)

Reply:
The application library needs a definition of the variables. The application library can exist without md-flex, md-flex can't exist without the application library. Hence, the variables need to be defined in the application library.
We could think about renaming them to eliminate the semantic association but honestly I don't care too much.

Move it to the application lib to get rid of the code duplication and let's get this fix merged asap.

Reply:
done in [fc2e569](https://github.com/AutoPas/AutoPas/pull/778/commits/fc2e569b5b29eae75b37dc5341b0fcb9b19cc58f)